hub/src/strategy/XChainStrategyStargate.sol

```diff
@@ -11,16 +11,17 @@
 // auxo.fi
 
 // SPDX-License-Identifier: AGPL-3.0-only
-pragma solidity 0.8.12;
+pragma solidity >=0.8.0;
 
-import {XChainStargateHub} from "../XChainStargateHub.sol";
-import {IVault} from "../interfaces/IVault.sol";
-import {BaseStrategy} from "./BaseStrategy.sol";
+import {XChainStargateHub} from "@hub/XChainStargateHub.sol";
+import {BaseStrategy} from "@hub/strategy/BaseStrategy.sol";
 
-import {IERC20} from "openzeppelin/token/ERC20/IERC20.sol";
-import {SafeERC20} from "openzeppelin/token/ERC20/utils/SafeERC20.sol";
+import {IVault} from "@interfaces/IVault.sol";
+import {IStargateReceiver} from "@interfaces/IStargateReceiver.sol";
+import {IERC20} from "@oz/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@oz/token/ERC20/utils/SafeERC20.sol";
 
-contract XChainStrategyStargate is BaseStrategy {
+contract XChainStrategyStargate is BaseStrategy, IStargateReceiver {
```

Comment:
The strategy should not be a IStargateReceiver. Only the hub is in charge of xchain communications